### PR TITLE
Create non-atomic views and separate transactions in core views

### PR DESCRIPTION
**Pull request by pawelkisielewicz**
Resolves https://github.com/golemfactory/concent/issues/782
It is updated and reopened https://github.com/golemfactory/concent/pull/834

### CODE REVIEW

**Comment by cameel** [comment]
@pawelkisielewicz When opening a new pull request for the code from a closed one please always link to the old one and explain that it's being reopened.

File: concent_api/common/tests/test_api_view.py [hunk 1]
```diff
@@ -56,31 +58,9 @@ def setUp(self):
...
-        task_to_compute = load(
-            dump(
-                task_to_compute,
-                REQUESTOR_PRIVATE_KEY,
-                settings.CONCENT_PUBLIC_KEY,
-            ),
-            settings.CONCENT_PRIVATE_KEY,
-            REQUESTOR_PUBLIC_KEY,
-            check_time=False,
-        )
```
Comment (pawelkisielewicz) [inline concent_api/common/tests/test_api_view.py]: It was not part of this task, but I found it, this block of code is not necessary. I deleted it, if I should'n please let me know

File: concent_api/core/message_handlers.py [hunk 2]
```diff
@@ -153,75 +154,76 @@ def handle_send_ack_report_computed_task(client_message: message.tasks.AckReport
...
-        if subtask.state_enum != Subtask.SubtaskState.FORCING_REPORT:
-            raise Http400(
-                "Subtask state is {} instead of FORCING_REPORT. Can't accept your 'AckReportComputedTask'.".format(
-                    subtask.state
-                ),
-                error_code=ErrorCode.QUEUE_WRONG_STATE,
-            )
+            if subtask.state_enum != Subtask.SubtaskState.FORCING_REPORT:
+                raise Http400(
+                    "Subtask state is {} instead of FORCING_REPORT. Can't accept your 'AckReportComputedTask'.".format(
```
Comment (kbeker) [inline concent_api/core/message_handlers.py]: Please change it to `f-string`

File: concent_api/core/subtask_helpers.py [hunk 3]
```diff
@@ -131,7 +132,13 @@ def update_timed_out_subtask(subtask: Subtask) -> None:
     )
 
 
-def update_timed_out_subtasks_in_message(client_message: Message, client_public_key: bytes) -> None:
+def update_subtasks_from_incoming_message_if_timed_out(client_message: Message, client_public_key: bytes) -> None:
+    """
+    Function gets subtask_id (or more subtask id's if message is ForcePayment) from client message, starts transaction,
+    checks if state is active and subtask is timed out (in database query, if it is subtask is locked). If so, file
+    status is verified (check additional conditions in verify_file_status) and subtask's state is updated
+
```
Comment (kbeker) [inline concent_api/core/subtask_helpers.py]: This blank is unnecessary.

File: concent_api/core/subtask_helpers.py [hunk 4]
```diff
@@ -141,37 +148,45 @@ def update_timed_out_subtasks_in_message(client_message: Message, client_public_
...
+            _update_timed_out_subtask(subtask)
 
 
-def update_all_timed_out_subtasks_of_client(client_public_key: bytes) -> None:
+def update_all_timed_out_subtasks_of_a_client(client_public_key: bytes) -> None:
+    """
+    Function looks for all subtasks in active state of client. All found subtasks are processed in separate transactions,
+    locked in database, file status is verified (check additional conditions in verify_file_status) and subtask's state
+    is updated in _update_timed_out_subtask
+
```
Comment (kbeker) [inline concent_api/core/subtask_helpers.py]: This blank is also unnecessary.

File: concent_api/core/subtask_helpers.py [hunk 5]
```diff
@@ -29,7 +30,7 @@
 logger = getLogger(__name__)
 
 
-def update_timed_out_subtask(subtask: Subtask) -> None:
+def _update_timed_out_subtask(subtask: Subtask) -> None:
```
Comment (kbeker) [inline concent_api/core/subtask_helpers.py]: This should not be in this commit because you are changing name of a function here.

File: concent_api/core/subtask_helpers.py [hunk 6]
```diff
@@ -131,7 +132,13 @@ def update_timed_out_subtask(subtask: Subtask) -> None:
     )
 
 
-def update_timed_out_subtasks_in_message(client_message: Message, client_public_key: bytes) -> None:
+def update_subtasks_from_incoming_message_if_timed_out(client_message: Message, client_public_key: bytes) -> None:
```
Comment (kbeker) [inline concent_api/core/subtask_helpers.py]: Also here. This change is not related with changes which you are doing in this PR

File: concent_api/core/subtask_helpers.py [hunk 7]
```diff
@@ -141,37 +148,45 @@ def update_timed_out_subtasks_in_message(client_message: Message, client_public_
...
+                return
 
-        verify_file_status(subtask=subtask, client_public_key=client_public_key)
-        update_timed_out_subtask(subtask)
+            verify_file_status(subtask=subtask, client_public_key=client_public_key)
+            _update_timed_out_subtask(subtask)
 
 
-def update_all_timed_out_subtasks_of_client(client_public_key: bytes) -> None:
+def update_all_timed_out_subtasks_of_a_client(client_public_key: bytes) -> None:
```
Comment (kbeker) [inline concent_api/core/subtask_helpers.py]: Also this function

File: concent_api/common/tests/test_api_view.py [hunk 8]
```diff
@@ -164,18 +166,18 @@ def dummy_view(_request, _message, _client_public_key):
...
+    raise Http400('', error_code=ErrorCode.MESSAGE_UNEXPECTED)
 
 
-def _update_timed_out_subtask_400_mock(_subtask_id, _client_public_key):
+def _create_client_and_raise_http500_error_mock(*_args, **_kwargs):
     Client.objects.get_or_create_full_clean(
         CONCENT_PUBLIC_KEY
     )
-    raise Http400('', error_code=ErrorCode.MESSAGE_UNEXPECTED)
+    raise TypeError
```
[the remark this answers is not in the document]
Comment (pawelkisielewicz) [reply concent_api/common/tests/test_api_view.py]: it is the same at it was in `_update_timed_out_subtask_500_mock`. I need it to generate HTTP500

File: concent_api/core/subtask_helpers.py (hunk 5 above)
Comment (rwrzesien) [reply concent_api/core/subtask_helpers.py]: Why this name was changed in general?

Comment (pawelkisielewicz) [reply concent_api/core/subtask_helpers.py]: I know but it is related with previous PR. It is not directly part of this task, but small improvement of previous. This change is not necessary, but after merging previous PR i decided it will be better as it is now. So i did it here, this is to small to separate PR and code looks better :)

File: concent_api/core/subtask_helpers.py (hunk 6 above)
Comment (pawelkisielewicz) [reply concent_api/core/subtask_helpers.py]: as above

File: concent_api/core/subtask_helpers.py (hunk 7 above)
Comment (pawelkisielewicz) [reply concent_api/core/subtask_helpers.py]: as above...

File: concent_api/core/message_handlers.py (hunk 2 above)
Comment (pawelkisielewicz) [reply concent_api/core/message_handlers.py]: it was only moved ... :D changed :)

File: concent_api/core/subtask_helpers.py (hunk 5 above)
Comment (pawelkisielewicz) [reply concent_api/core/subtask_helpers.py]: It is changed because this function is used only in `subtask_helpers.py`, called by two other functions in this file, so i wanted to mark it as 'private'

**Comment by pawelkisielewicz** [comment]
It is not, because the lot of changed lines is remove `transaction.atomic` decorator from view and add new atomic blocks in message_handlers. It is one logic change, if it would be couple of commits in this commits application would not be in "working" state. Besides, most of "changed" lines is only moving by 4 spaces because of context manager.

File: concent_api/common/tests/test_api_view.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (dybi) [reply concent_api/common/tests/test_api_view.py]: it should not be aa `TypeError` - it si falsely suggesting that this particular exception is needed here. It should be either `Exception` or some  custom exception inheriting from it.
